pinecone/pinecone.py

```diff
@@ -0,0 +1,327 @@
+import logging
+from typing import Optional, Dict, Union, TYPE_CHECKING
+from multiprocessing import cpu_count
+import warnings
+
+from pinecone.config import PineconeConfig, ConfigBuilder
+
+from .legacy_pinecone_interface import LegacyPineconeDBControlInterface
+
+from pinecone.utils import normalize_host, PluginAware, docslinks
+from .langchain_import_warnings import _build_langchain_attribute_error_message
+
+logger = logging.getLogger(__name__)
+""" @private """
+
+if TYPE_CHECKING:
```

Comment:
Moving these imports under this `if TYPE_CHECKING` means they won't be loaded except during the static type analysis process. A big performance savings during initialization at runtime.  Many of these things used to be needed here at runtime but no longer are because functionality has been pulled out into classes like `IndexResource`.